independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java

```diff
@@ -13,10 +13,24 @@
 public final class InitialConfigurator implements EmbeddedConfigurator {
 
     public static final QuarkusDelayedHandler DELAYED_HANDLER = new QuarkusDelayedHandler();
+    private static final Level MIN_LEVEL;
+
+    static {
+        Level minLevel = Level.ALL;
+        String minLevelSysProp = System.getProperty("logging.initial-configurator.min-level");
+        if (minLevelSysProp != null) {
+            try {
+                minLevel = Level.parse(minLevelSysProp);
+            } catch (IllegalArgumentException ignored) {
```

Comment:
Should we log a warning here as well?

Reply:
Not sure we can from inside a piece of code that configures logging :). I guess we could do `System.err`, WDYT?

Reply:
A fatal error could be suitable?

BTW are you sure the default should be `ALL` ? I thought it was debug, effectively having tracing off.

Reply:
Yes, that was my thought...

Reply:
I don't want to change the default in this PR, we can certainly do that in some other purpose built PR.

Dunno about a fatal error, we could do that to - I don't have a strong opinion (as it's pretty much impossible this situation will arise).

Reply:
PR updated to turn the parsing exception into a terminal failure

Reply:
> I don't want to change the default in this PR, we can certainly do that in some other purpose built PR.

What I'm saying is that `ALL` is NOT the default today

Reply:
Apparently the default is `DEBUG`:
 - https://quarkus.io/guides/all-config#quarkus-core_quarkus.log.min-level

It's important because we want `TRACE` to be disabled by default.

Reply:
`ALL` is currently the default for this piece of code: https://github.com/quarkusio/quarkus/blob/2.12.1.Final/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java#L19

We can certainly align the log level with the quarkus min level default, but I would prefer doing that in another PR, which only does that one thing

Reply:
Ok I certainly do like small patches.

Still it's confusing me.. wasn't the point of this patch to actually honour the `min-level` configured value?

That's what I would expect from the issue description.

Reply:
Yes, and this is what the code does - there is no chance the default codepath can be taken as the system property is always set.
But we can certainly change the default to `DEBUG` as well in the next iteration of this

Reply:
> the system property is always set

oh really? can I read any build time property this way? I didn't know

Reply:
ah sorry, I see now the other builditem is setting it. Missed that, ok ignore me then :+1: 

Thanks!

Reply:
Well, it's a hack :)

We set a custom system property based on the build time property and then make sure we read it here - this works because these custom system properties are set extremely early.